src/range_proof.rs

```diff
@@ -674,20 +682,26 @@ where
             let li = proof.li_decompressed()?;
             let ri = proof.ri_decompressed()?;
 
+            // Useful lengths
+            let aggregation_factor = commitments.len();
+            let full_length = aggregation_factor
+                .checked_mul(bit_length)
+                .ok_or(ProofError::SizeOverflow)?;
+            let rounds = li.len();
+
             if li.len() != ri.len() {
                 return Err(ProofError::InvalidLength(
                     "Vector L length not equal to vector R length".to_string(),
                 ));
             }
-            if 1 << li.len() != commitments.len() * bit_length {
-                return Err(ProofError::InvalidLength("Vector L length not adequate".to_string()));
+            if 1usize
+                .checked_shl(u32::try_from(rounds).map_err(|_| ProofError::SizeOverflow)?)
+                .ok_or(ProofError::SizeOverflow)? !=
+                full_length
+            {
+                return Err(ProofError::InvalidLength("Vector L/R length not adequate".to_string()));
```

Comment:
See https://github.com/tari-project/tari/pull/5664 - `tari/docs/src/reviewing_guide.md` - paragraph `Shifts (<<, >>)`
```suggestion
            let rounds_u32 = u32::try_from(rounds).map_err(|_| ProofError::SizeOverflow)?;
            if rounds_u32.leading_zeros() < 1 {
                return Err(ProofError::SizeOverflow);
            }
            if 1usize.checked_shl(rounds_u32).ok_or(ProofError::SizeOverflow)? != full_length {
                return Err(ProofError::InvalidLength("Vector L/R length not adequate".to_string()));
```

Reply:
What an infuriatingly inconsistent way for those shift functions to behave! Will update.